src/routes/status.tsx

```diff
@@ -5,22 +5,287 @@
+    stake: string;
+    0: string;
+    1: string;
+    2: string;
+    3: boolean;
+    4: string;
+    5: string;
+  };
+  suspect: boolean;
+}
 
+@inject('exitHandler', 'explorer', 'web3', 'tokens', 'bridge')
 @observer
-export default class Status extends React.Component {
+export default class Status extends React.Component<StatusProps, {}> {
+  private exitColumns: ColumnProps<any>[] = [
+    {
+      title: 'Amount',
+      dataIndex: 'exit.amount',
+      key: 'amount',
+      render: (value, record) => (
+        <TokenValue value={value} color={record.exit.color} />
+      ),
+    },
+    {
+      title: 'Owner',
+      dataIndex: 'exit.owner',
+      key: 'owner',
+    },
+    {
+      title: 'PriorityTimestamp',
+      dataIndex: 'exit.priorityTimestamp',
+      key: 'priorityTimestamp',
+      render: text => {
+        return <p>{new Date(parseInt(text) * 1000).toUTCString()}</p>;
+      },
+    },
+    {
+      title: 'Stake',
+      dataIndex: 'exit.stake',
+      key: 'stake',
+      render: (value, record) =>
+        `${this.props.web3.root.instance.utils.fromWei(value, 'ether')} ETH`,
+    },
+    {
+      title: 'Suspicious',
+      dataIndex: 'suspect',
+      key: 'suspect',
+      render: suspect => {
+        if (suspect) return <p style={{ color: 'red' }}>Yes</p>;
+        else return <p style={{ color: 'green' }}>No</p>;
+      },
+    },
+  ];
+
+  private colorColums: ColumnProps<any>[] = [
+    {
+      title: 'Color',
+      dataIndex: 'color',
+      key: 'color',
+    },
+    {
+      title: 'Plasma balance',
+      dataIndex: 'plasmaBalance',
+      key: 'plasmaBalance',
+      render: (value, record) => (
+        <TokenValue value={value} color={record.color} />
+      ),
+    },
+    {
+      title: 'Utxo Sum',
+      dataIndex: 'utxoSum',
+      key: 'utxoSum',
+      render: (value, record) => (
+        <TokenValue value={value} color={record.color} />
+      ),
+    },
+    {
+      title: 'Open Exit Sum',
+      dataIndex: 'exitSum',
+      key: 'exitSum',
+      render: (value, record) => (
+        <TokenValue value={value} color={record.color} />
+      ),
+    },
+    {
+      title: 'Status',
+      key: 'status',
+      render: (value, record: ColorDetails) => {
+        return record.plasmaBalance.gte(record.exitSum.add(record.utxoSum)) ? (
+          <Badge status="success" text="Ok" />
+        ) : (
+          <Badge status="warning" text="Invalid" />
+        );
+      },
+    },
+  ];
+
+  constructor(props) {
+    super(props);
+    reaction(() => this.props.exitHandler.address, this.getData, {
+      fireImmediately: true,
+    });
+  }
+
+  @observable exitDetails: ExitData[] = [];
+  @observable colorDetails: ColorDetails[] = [];
+
+  @action
+  private getData = async (): Promise<void> => {
+    const { exitHandler, explorer, web3, tokens, bridge } = this.props;
+    const exits = [];
+    if (!bridge.address)
+      await new Promise(resolve => reaction(() => bridge.address, resolve));
```

Comment:
Seems strange. If you need exitHandler and bridge in this method, will be better to use `autorun`.

**getData**
```ts
private getData = async (): Promise<void> => {
  if (!bridge.address || !exitHandler.address) {
    return;
  }
  ... do fetching here
}
```

**constructor**
```ts
constructor(props) {
  super(props);
  autorun(this.getData);
}
```